sdk/com.ibm.sbt.web/src/main/webapp/js/sdk/_bridges/dojo-amd/RequestTransport.js

```diff
@@ -55,7 +55,19 @@ define([ "dojo/_base/declare", "dojo/_base/lang", "dojo/request", "../util" ], f
          * 
          */
         request : function(url,options) {
-            return request(url, options);
+        	var promise = new Promise();
+            promise.response = new Promise();
+            request(url, args).response.then(
+                function(response){
```

Comment:
@projsaha On your first point. The Return type of RequestTransport.js and Transport.js are the same and with these changes consumers can switch between the two, however the code that is being used at the moment does not work with the sdk, because it does not return a promise. The Dojo Request module returns a promise that has a property called response. The createResponse function in Transport.js creates this response property but since dojo   returns the response property automatically there is no need to call createResponse and the return types of both Transport.js and RequestTransport.js will be the same. 

This Api is not being used anywhere in the toolkit, if it was it would break the toolkit in its current implementation because it is not returning a promise but rather a function. I have tried it in the project i am working on now and this implementation is working and i can also easily switch between the two transport modules .